### ISSUE TITLE
Force basic auth on first request to DAV-backend

### ISSUE THREAD

**Issue by kearva** [blind-coder/rcmcarddav#309]
My backend uses basic auth. It would be nice to have a configuration option send logon information on first request. It would probably halve the number of requests to the DAV-backend.

**Comment by mstilkerich** [comment]
I'd like to keep things simple and not introduce and especially maintain features / configuration variants that I don't see a real need for. Technically, you could save a request. However, the HTTP connection is reused and therefore the expense of a TCP handshake to establish a new connection is paid only once. I don't think this extra request to query the offered authentication mechanisms therefore causes a significant overhead in the overall communication of a sync operation.

Therefore, unless you have a very good reason / actual problem with the current behavior, I would rather not implement this.

(btw rcmcarddav used to have such an option, but it was from a time where a very basic HTTP client lib was used that created a new connection for each request, and AFAIR also an interoperability issue with some CardDAV server though I cannot remember the details)

**Comment by kearva** [comment]
Depending on number of contacts, time between syncs and rtt between RoundCube and the CardDav-server it might result in a slower performance.

But at the same time I wouldn't lose any sleep if this isn't implemented. It is more a nice-to-have rather than a demand ;-)

**Comment by mstilkerich** [comment]
Well, the connection is reused, so you only have the extra roundtrip once per sync for the very first request, for all following requests of the sync operation the auth header is sent immediately. The more data is transferred during the sync, the less this extra roundtrip causes relative to the entire sync operation. I don't think you will notice any difference, syncs with no changes (the most common case) feel instant with all major services I tried (< 1 second).
